Ffeldberg/rel 8140/add view features to tf provider (#329)
launchdarkly/terraform-provider-launchdarkly: docs/data-sources/feature_flag.md

## Example Usage

```terraform

> # Example: Get feature flag information including view memberships

Reviewer: nit, optional to address: I'm not sure we use "memberships" elsewhere (yet) to describe this. Does this mean the views that a flag is linked to? Maybe something like this?

```suggestion
# Example: Get feature flag information including linked views
```

data "launchdarkly_feature_flag" "example" {
  project_key = "example-project"
  key         = "example-flag"
}

> # The feature flag data source now includes discovery of view memberships

Reviewer: ```suggestion
# The feature flag data source now includes discovery of linked views
```
(only if you also choose to use the previous suggestion)

output "flag_details" {
  value = {
    name        = data.launchdarkly_feature_flag.example.name
    description = data.launchdarkly_feature_flag.example.description
    views       = data.launchdarkly_feature_flag.example.views
    archived    = data.launchdarkly_feature_flag.example.archived
  }
}

# Example: Check if flag is accessible to specific teams
locals {
  accessible_to_frontend = contains(data.launchdarkly_feature_flag.example.views, "frontend-team")
  accessible_to_mobile   = contains(data.launchdarkly_feature_flag.example.views, "mobile-team")
  is_shared_flag         = length(data.launchdarkly_feature_flag.example.views) > 1
}

> # Example: Conditional resource creation based on view membership

Reviewer: ```suggestion
# Example: Conditional resource creation based on views
```
optional

resource "launchdarkly_feature_flag_environment" "prod_config" {
  # Only create production config if flag is assigned to production view
  count = contains(data.launchdarkly_feature_flag.example.views, "production-ready") ? 1 : 0

  flag_id = data.launchdarkly_feature_flag.example.id
  env_key = "production"
  on      = true
}

> # Example: Generate team notifications based on flag view assignments

Reviewer: ```suggestion
# Example: Generate team notifications based on views
```
optional
